examples/sugarscape_g1mt/sugarscape_g1mt/agents.py

```diff
@@ -0,0 +1,334 @@
+import math
+import numpy as np
+
+import mesa
+
+# Note: GAS refers to the book by Epstein, "Growing Artificial Societies:
+# Social Science from the Bottom Up," published in 1996.
+
+
+def get_distance(pos_1, pos_2):
+    """Calculate Euclidean distance between two positions.
+    Args:
+        pos_1, pos_2: Coordinate tuples for both points.
+    """
+    x1, y1 = pos_1
+    x2, y2 = pos_2
+    dx = x1 - x2
+    dy = y1 - y2
+    return math.sqrt(dx**2 + dy**2)
+
+
+class Sugar(mesa.Agent):
+    """
+    Sugar is a FSM that
+    - contains an amount of sugar
+    - grows 1 amount of sugar at each turn (Epstein's rule G1).
+    """
+
+    def __init__(self, unique_id, model, pos, max_sugar):
+        super().__init__(unique_id, model)
+        self.pos = pos
+        self.amount = max_sugar
+        self.max_sugar = max_sugar
+
+    def step(self):
+        self.amount = min([self.max_sugar, self.amount + 1])
+
+
+class Spice(mesa.Agent):
+    """
+    Spice is a FSM that
+    - contains an amount of spice
+    - grows 1 amount of spice at each turn (Epstein's rule G1).
+    """
+
+    def __init__(self, unique_id, model, pos, max_spice):
+        super().__init__(unique_id, model)
+        self.pos = pos
+        self.amount = max_spice
+        self.max_spice = max_spice
+
+    def step(self):
+        self.amount = min([self.max_spice, self.amount + 1])
+
+
+class Trader(mesa.Agent):
+    def __init__(
+        self,
+        unique_id,
+        model,
+        pos,
+        moore=False,
+        sugar=0,
+        spice=0,
+        metabolism_sugar=0,
+        metabolism_spice=0,
+        vision=0,
+    ):
+        super().__init__(unique_id, model)
+        self.pos = pos
+        self.moore = moore
+        self.sugar = sugar
+        self.spice = spice
+        self.metabolism_sugar = metabolism_sugar
+        self.metabolism_spice = metabolism_spice
+        self.vision = vision
+
+        self.prices = []
+        self.trade_partners = []
+
+    def get_sugar(self, pos):
+        this_cell = self.model.grid.get_cell_list_contents(pos)
+        for agent in this_cell:
+            if type(agent) is Sugar:
+                return agent
+        return None
+
+    def get_sugar_amount(self, pos):
+        sugar_patch = self.get_sugar(pos)
+        if sugar_patch:
+            return sugar_patch.amount
+        return 0
+
+    def get_spice(self, pos):
+        this_cell = self.model.grid.get_cell_list_contents([pos])
+        for agent in this_cell:
+            if isinstance(agent, Spice):
+                return agent
+        return None
+
+    def get_spice_amount(self, pos):
+        spice_patch = self.get_spice(pos)
+        if spice_patch:
+            return spice_patch.amount
+        return 0
+
+    def get_trader(self, pos):
+        this_cell = self.model.grid.get_cell_list_contents(pos)
+        for agent in this_cell:
+            if isinstance(agent, Trader):
+                return agent
+
+    def is_occupied(self, pos):
```

Comment:
@tpike3 what about this explanatory comment? With this, we can clearly separate the condition. What do you think of renaming `is_occupied` to `is_occupied_by_other`?
```python
if pos == self.pos:
    # We consider the agent's own position to be "unoccupied" in the sense that it is one of the legal positions the agent can occupy.
    return False
this_cell = self.model.grid.get_cell_list_contents(pos)
for a in this_cell:
    if isinstance(a, Trader):
        # Occupied by other agent.
        return True
return False
```

Reply:
That works